File: app/src/main/java/com/nibou/niboucustomer/customviews/MySlidingImageAdapter.java
package com.nibou.niboucustomer.customviews;

import android.content.Context;
import android.os.Parcelable;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ProgressBar;

import com.bumptech.glide.Glide;
import com.bumptech.glide.load.resource.drawable.GlideDrawable;
import com.bumptech.glide.request.RequestListener;
import com.bumptech.glide.request.target.Target;
import com.nibou.niboucustomer.R;
import com.nibou.niboucustomer.models.ListResponseModel;

import java.util.ArrayList;

public class MySlidingImageAdapter extends PagerAdapter {

    ArrayList<ListResponseModel.Img> imgArrayList;
    private LayoutInflater inflater;
    private Context context;


    public MySlidingImageAdapter(Context context, ArrayList<ListResponseModel.Img> imgArrayList) {
        this.context = context;
        this.imgArrayList = imgArrayList;
        inflater = LayoutInflater.from(context);
    }

    @Override
    public int getCount() {
        return imgArrayList.size();
    }

    @Override
    public Object instantiateItem(ViewGroup view, final int position) {
        View imageLayout = inflater.inflate(R.layout.layout_swipe_image, view, false);
        assert imageLayout != null;

        final ImageView imageView = (ImageView) imageLayout.findViewById(R.id.zoomableImageView);

        imageView.setOnClickListener(v -> {
            if (context instanceof MyImageViewPagerActivity) {
                ((MyImageViewPagerActivity) context).handleCrossButton();
            }
        });

        final ProgressBar progressBar = (ProgressBar) imageLayout.findViewById(R.id.progressBar);
        progressBar.setVisibility(View.VISIBLE);
        if (imgArrayList.get(position).getImg() != null && !imgArrayList.get(position).getImg().isEmpty()) {
            Glide.with(context).load(imgArrayList.get(position).getImg()).error(R.drawable.empty_image_theme_color_drawable).placeholder(R.drawable.empty_image_theme_color_drawable).listener(new RequestListener<String, GlideDrawable>() {
                @Override
                public boolean onException(Exception e, String model, Target<GlideDrawable> target, boolean isFirstResource) {
                    return false;
                }

                @Override
                public boolean onResourceReady(GlideDrawable resource, String model, Target<GlideDrawable> target, boolean isFromMemoryCache, boolean isFirstResource) {
                    progressBar.setVisibility(View.GONE);
                    return false;
                }
            }).into(imageView);
        }
        view.addView(imageLayout, 0);
        return imageLayout;
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view.equals(object);
    }

    @Override
    public void restoreState(Parcelable state, ClassLoader loader) {
    }

    @Override
    public Parcelable saveState() {
        return null;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((View) object);
    }
}

File: app/src/main/java/com/nibou/niboucustomer/customviews/MySlidingImageAdapter.kt
//package com.ongraph.muslimhub.ui.activities
//
//import android.content.Context
//import android.graphics.drawable.Drawable
//import android.net.Uri
//import android.os.Parcelable
//import android.support.v4.view.PagerAdapter
//import android.view.LayoutInflater
//import android.view.View
//import android.view.ViewGroup
//import android.widget.ImageView
//import android.widget.ProgressBar
//import com.bumptech.glide.Glide
//import com.bumptech.glide.load.DataSource
//import com.bumptech.glide.load.engine.DiskCacheStrategy
//import com.bumptech.glide.load.engine.GlideException
//import com.bumptech.glide.request.RequestListener
//import com.bumptech.glide.request.RequestOptions
//import com.bumptech.glide.request.target.Target
//import com.ongraph.muslimhub.R
//
//class MySlidingImageAdapter(private val context: Context, internal var mImageList: List<String>) : PagerAdapter() {
//    private val inflater: LayoutInflater = LayoutInflater.from(context)
//
//    override fun getCount(): Int {
//        return mImageList.size
//    }
//
//    override fun instantiateItem(view: ViewGroup, position: Int): Any {
//        val imageLayout = inflater.inflate(R.layout.layout_swipe_image, view, false)!!
//        val imageView = imageLayout.findViewById(R.id.zoomableImageView) as ImageView
//
//        val progressBar = imageLayout.findViewById(R.id.progressBar) as ProgressBar
//
//        progressBar.visibility = View.VISIBLE
//        if (mImageList[position] != null) {
//            val imageUri = Uri.parse(mImageList[position])
//
//            val requestOptions = RequestOptions()
//                    .placeholder(R.mipmap.default_header).error(R.mipmap.default_header)
//                    .diskCacheStrategy(DiskCacheStrategy.AUTOMATIC)
//
//            Glide.with(context)
//                    .load(imageUri)
//                    .listener(object : RequestListener<Drawable> {
//                        override fun onLoadFailed(e: GlideException?, model: Any, target: Target<Drawable>, isFirstResource: Boolean): Boolean {
//                            progressBar.visibility = View.GONE
//                            return false
//                        }
//
//                        override fun onResourceReady(resource: Drawable, model: Any, target: Target<Drawable>, dataSource: DataSource, isFirstResource: Boolean): Boolean {
//                            progressBar.visibility = View.GONE
//                            return false
//                        }
//                    })
//                    .apply(requestOptions)
//                    .into(imageView)
//                    .clearOnDetach()
//        }
//        view.addView(imageLayout, 0)
//        return imageLayout
//    }
//
//    override fun isViewFromObject(view: View, `object`: Any): Boolean {
//        return view == `object`
//    }
//
//    override fun restoreState(state: Parcelable?, loader: ClassLoader?) {}
//
//    override fun saveState(): Parcelable? {
//        return null
//    }
//
//    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
//        container.removeView(`object` as View)
//    }
//}